core/src/main/scala/za/co/absa/fadb/status/package.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.fadb
+
+import za.co.absa.fadb.exceptions.StatusException
+
+package object status {
+
+  /**
+    *  Class represents the status of calling a fa-db function (if it supports status that is)
+    */
+  case class FunctionStatus(statusCode: Int, statusText: String)
+
+  /**
+    *  Represents a function status with data.
+    *  @param functionStatus the function status
+    *  @param data the data of one row (barring the status fields)
+    *  @tparam D the type of the data
+    */
+  case class Row[D](functionStatus: FunctionStatus, data: D)
```

Comment:
I am not sure if `Row` is a good name. Also why `D` for the type parameter?

Reply:
`D` simply says that it's the `data`, or, 'effective/actual data' (sometimes this term can be find in literature). So basically a row without the status-related columns.

Hmm, why not? `Row` indicates that 1 instance of this should represent 1 row being returned from the DB. I had the name much more complicated, and it's true that `Row` is pretty simple, naive even, but when I checked the usecases and thought about its internal structure, I think that it's good enough name for this (couldn't come up with anything better and I didn't want the names to be sentences :D )

Also, this naming was not chosen to be 'isolated' to this `Row` class - we wanted to make it kind of relatable and consistent with the rest of the case classes in this module - see the classes below in this module :)  

But as always, I'm open to proposals